config/fs.go

```diff
@@ -0,0 +1,239 @@
+type TimeoutConfig struct {
+	// DialTimeout is the maximum duration that connection can take before a request attempt is timed out.
+	DialTimeoutMsec int64
+	// ResponseHeaderTimeout is the maximum duration waiting for response headers before a request attempt is timed out.
+	// This starts after the entire request body is uploaded to the remote endpoint and stops when the request headers
+	// are fully read. It does not include reading the body.
+	ResponseHeaderTimeoutMsec int64
+	// RequestTimeout is the maximum duration before the entire request attempt is timed out. This starts when the
+	// client starts the connection attempt and ends when the entire response body is read.
+	RequestTimeoutMsec int64
+}
+
+// RetryableHTTPClientConfig is the complete config for a retryable http client
+type RetryableHTTPClientConfig struct {
+	TimeoutConfig
+	RetryConfig
+}
+
+func parseFSConfig(cfg *Config) {
+	if cfg.FuseConfig.AttrTimeout == 0 {
+		cfg.FuseConfig.AttrTimeout = defaultFuseTimeoutSec
+	}
+
+	if cfg.FuseConfig.EntryTimeout == 0 {
+		cfg.FuseConfig.EntryTimeout = defaultFuseTimeoutSec
+	}
+
+	if cfg.FuseConfig.NegativeTimeout == 0 {
+		cfg.FuseConfig.NegativeTimeout = defaultFuseTimeoutSec
+	}
+
+	if cfg.BackgroundFetchConfig.FetchPeriodMsec == 0 {
+		cfg.BackgroundFetchConfig.FetchPeriodMsec = defaultBgFetchPeriodMsec
+	}
+
+	if cfg.BackgroundFetchConfig.SilencePeriodMsec == 0 {
+		cfg.BackgroundFetchConfig.SilencePeriodMsec = defaultBgSilencePeriodMsec
+	}
+
+	if cfg.BackgroundFetchConfig.MaxQueueSize == 0 {
+		cfg.BackgroundFetchConfig.MaxQueueSize = defaultBgMaxQueueSize
+	}
+
+	if cfg.BackgroundFetchConfig.EmitMetricPeriodSec == 0 {
+		cfg.BackgroundFetchConfig.EmitMetricPeriodSec = defaultBgMetricEmitPeriodSec
+	}
+
+	if cfg.MountTimeoutSec == 0 {
+		cfg.MountTimeoutSec = defaultMountTimeoutSec
+	}
+
+	if cfg.FuseMetricsEmitWaitDurationSec == 0 {
+		cfg.FuseMetricsEmitWaitDurationSec = defaultFuseMetricsEmitWaitDurationSec
+	}
+
+	// RetryableHTTPClientConfig
+
+	if cfg.RetryableHTTPClientConfig.TimeoutConfig.DialTimeoutMsec == 0 {
+		cfg.RetryableHTTPClientConfig.TimeoutConfig.DialTimeoutMsec = defaultDialTimeoutMsec
+	}
+
+	if cfg.RetryableHTTPClientConfig.TimeoutConfig.ResponseHeaderTimeoutMsec == 0 {
+		cfg.RetryableHTTPClientConfig.TimeoutConfig.ResponseHeaderTimeoutMsec = defaultResponseHeaderTimeoutMsec
+
+	}
+	if cfg.RetryableHTTPClientConfig.TimeoutConfig.RequestTimeoutMsec == 0 {
+		cfg.RetryableHTTPClientConfig.TimeoutConfig.RequestTimeoutMsec = defaultRequestTimeoutMsec
+
+	}
+	if cfg.RetryableHTTPClientConfig.RetryConfig.MaxRetries == 0 {
+		cfg.RetryableHTTPClientConfig.RetryConfig.MaxRetries = defaultMaxRetries
+
+	}
+	if cfg.RetryableHTTPClientConfig.RetryConfig.MinWaitMsec == 0 {
+		cfg.RetryableHTTPClientConfig.RetryConfig.MinWaitMsec = defaultMinWaitMsec
+
+	}
+	if cfg.RetryableHTTPClientConfig.RetryConfig.MaxWaitMsec == 0 {
+		cfg.RetryableHTTPClientConfig.RetryConfig.MaxWaitMsec = defaultMaxWaitMsec
+	}
```

Comment:
Same question as for BlobConfit. Separate constructors would be more readable and reusable.